.storybook/decorators/PropCombinations/PropCombinationsDecorator.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ CAUTION ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Updating this file will flag a change
+ * in _every_ component that leverages generated stories.
+ *
+ * Modify with caution
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+import React from 'react';
+import { Decorator, StoryContext, StoryFn } from '@storybook/react';
+import { entries, isUndefined } from 'lodash';
+
+import {
+  StoryType,
+  type GeneratedStoryConfig,
+  type StoryMetaType,
+} from '@leafygreen-ui/lib';
+
+import { generatedStoryWrapper } from './PropCombinations.styles';
+import { Err, PropCombinations } from './components';
+import { isGeneratedStory } from './utils';
+import { PARAM_NAME } from './constants';
+
+type ContextType<T extends React.ComponentType<any>> = StoryContext<T> &
+  StoryMetaType<T>;
+
+const PropCombinationsDecorator: Decorator = (
+  StoryFn: StoryFn,
+  context: StoryContext<unknown>,
+) => {
+  const { component } = context;
+
+  if (component) {
+    const { parameters, args: metaArgs } = context as ContextType<
+      typeof component
+    >;
+
+    const decoratorConfig: GeneratedStoryConfig<typeof component> | undefined =
+      parameters[PARAM_NAME];
+
+    if (!isUndefined(decoratorConfig)) {
+      const {
+        props,
+        excludeCombinations,
+        args: generatedArgs,
+        decorator,
+      } = decoratorConfig;
+      if (isGeneratedStory(context)) {
+        // Check for props
+        if (!props || entries(props).length === 0) {
+          return Err('`props` not found in story generation parameters.');
+        }
+
+        // Remove any args that are explicitly defined
+        for (let propName in { ...generatedArgs }) {
+          if (props[propName]) {
```

Comment:
this looks like we're letting args overwrite props, not the other way around (which the comment implies)